ddl/column.go

```diff
@@ -308,6 +318,39 @@ func doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldN
 	// }
 	// }
 
+	if !mysql.HasNotNullFlag(oldCol.Flag) && mysql.HasNotNullFlag(newCol.Flag) {
+		// Get sessionctx from context resource pool.
+		var ctx sessionctx.Context
+		ctx, err = w.sessPool.get()
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		defer w.sessPool.put(ctx)
+
+		// Modify the type defined Flag to NotNullFlag.
+		tblInfo.Columns[oldCol.Offset].Flag = newCol.Flag
+		ver, err = updateVersionAndTableInfo(t, job, tblInfo, true)
+		// Wait for two leases to ensure that all nodes in the cluster are updated successfully.
+		ver, err = updateVersionAndTableInfo(t, job, tblInfo, true)
```

Comment:
Why update it again? What if tidb panics after this?

Reply:
This question needs to be discussed.